### PR TITLE
Transform and insert data into national or hackney addresses

### PR DESCRIPTION
**Pull request by elcorbs**
This new lambda will get records from the address base table. You can specify a limit, offset and whether to only get hackney/ non hackney addresses. 
Then it will format them to fit into the other tables and them then into the hackney or national address tables.

You can only run for hackney or national, the default is hackney.

Also Moved tests around  DatabaseActions into a separate file to the test database setup, so that could be used for handler tests.

**This is pretty rough so please give suggestions for improvement/ cleaning up**

### CODE REVIEW

File: AddressesDataPipeline/Database/DatabaseActions.cs [hunk 1]
```diff
@@ -28,9 +29,34 @@ public int CopyDataToDatabase(string tableName, ILambdaContext context, string a
...
+            var insertStatement = $"INSERT INTO dbo.{databaseToInsertInto}" +
+              "(lpi_key,uprn,usrn,parent_uprn,lpi_logical_status,sao_text,pao_text,building_number,street_description," +
+              "postcode,postcode_nospace,locality,gazetteer,organisation,ward,usage_description,usage_primary,blpu_class," +
+              "planning_use_class,property_shell,neverexport,easting,northing,longitude," +
+              "latitude,lpi_start_date,lpi_end_date,lpi_last_update_date,blpu_start_date,blpu_end_date,blpu_last_update_date," +
+              "line1,line2,line3,line4,town)" +
+              "VALUES ";
+
+            var values = string.Join(", ", records.Select(x =>
+                $"('{x.id:D14}' ,{x.uprn}, {x.usrn}, {x.parent_uprn}, 'Approved Preferred', '{x.sub_building}'," +
```
Comment (Renny4Real) [inline AddressesDataPipeline/Database/DatabaseActions.cs:51]: Nice 🎉

**Review by Renny4Real — APPROVED** [review]
Looks good! 😄

**Review by makeyw — APPROVED** [review]
All very excellent stuff.
You make me feel like I should give up coding! :) 
Let's try this tomorrow.

File: AddressesDataPipeline.sln.DotSettings.user [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:String x:Key="/Default/Environment/UnitTesting/UnitTestSessionStore/Sessions/=0f2583ff_002D85f2_002D4b26_002Dbf03_002Dc607ff6eb6b2/@EntryIndexedValue">&lt;SessionState ContinuousTestingIsOn="False" ContinuousTestingMode="0" FrameworkVersion="{x:Null}" IsLocked="False" Name="All tests from Solution" PlatformMonoPreference="{x:Null}" PlatformType="{x:Null}" xmlns="urn:schemas-jetbrains-com:jetbrains-ut-session" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"&gt;
```
Comment (humulla) [inline AddressesDataPipeline.sln.DotSettings.user]: Never seen this before, what does this do?

File: AddressesDataPipeline/serverless.yml [hunk 3]
```diff
@@ -5,6 +5,15 @@ provider:
   runtime: dotnetcore3.1
   vpc: ${self:custom.vpc.${opt:stage}}
   stage: ${opt:stage}
+  package:
+    artifact: bin/release/netcoreapp3.1/AddressesDataPipeline.zip
+  environment:
+    DB_HOST: ${ssm:/addresses-api/${self:provider.stage}/postgres-hostname}
```
Comment (humulla) [inline AddressesDataPipeline/serverless.yml:11]: Are new parameter store going to be created for these, or are we linking it to the existing one?

**Review by humulla — APPROVED** [review]
🎉🎉 Amazing work!
Just added some comments for my own knowledge

File: AddressesDataPipeline.sln.DotSettings.user (hunk 2 above)
Comment (elcorbs) [reply AddressesDataPipeline.sln.DotSettings.user]: This was added by resharper, doesn't need to be in here so I've removed it. Thanks for spotting!

File: AddressesDataPipeline/serverless.yml (hunk 3 above)
Comment (elcorbs) [reply AddressesDataPipeline/serverless.yml:11]: These are already in parameter store. Moving them into this section of the serverless file will make them available to both lambda functions in declared here. Before they were inside one of the lambda functions definition

**Pull request merged**